src/code/z_sub_s.c

```diff
@@ -214,17 +213,252 @@ void SubS_UpdateFlags(u16* flags, u16 setBits, u16 unsetBits) {
     *flags = (*flags & ~unsetBits) | setBits;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sub_s/func_8013AF00.s")
+/**
+ * Fills the weightArray to be used with time paths
+ *
+ * @param weightArray an array of values that are used to compute the weightVal and the individual weights
+ * @param numPoints the number of points considered with weights
+ * @param len the length to fill the array, generally the path count + numPoints
+ *
+ * @note `SubS_TimePathing_Update` assumes numPoints to be 3
+ */
+void SubS_TimePathing_FillWeightArray(f32 weightArray[], s32 numPoints, s32 len) {
+    s32 i;
+    f32 val = 0.0f;
+
+    for (i = 0; i < len; i++) {
+        if ((i >= numPoints) && (i < (len - numPoints + 1))) {
+            val += 1.0f;
+        }
+        weightArray[i] = val;
+    }
```

Comment:
I'm not sure about renaming `weightArray` to `offsetIndices`. I agree that represents the data well, but the same thing is used in calculating the interpolation weights which I don't think makes sense using something called `offsetIndices` (I could be wrong I don't know too much about interpolations and spline). Personally I think it makes more sense to base it off the interpolation usage, rather than from this function.
Also of note is that the time pathing doesn't necessarily require this particular `array`, it is just the default one and the only one that eventually got used.

Reply:
I see, yeah that's fine then. I wonder if it's worth adding something in the function itself then, just a comment explaining the for-loop and how it's creating an array of indices?

Reply:
Yeah, I can add a more descriptive description to the function itself explaining what the array looks like.